SEED-platform/seed
# 3218 feat/refresh metadata

#### Any background context you want to provide?
The OEP connection required the property/taxlot metadata to be refreshed to the current date to enable a record to be sync'd with salesforce

#### What's this PR do?
This PR adds an inventory list 'action' that allows a user to change the selected properties' "updated" timestamp attribute to the current time.

The user will be shown a progress bar as the celery task executes. Upon completion the modal will close and the page will refresh.

#### How should this be manually tested?
To test user experience: 
inventory > select a number of properties > actions > refresh metadata > refresh_metadata > watch progress bar go to 100 and modal will close 

To check actual property values 'updated' attribute (more involved):
```
docker exec -it seed_web /bin/bash
python manage.py shell
from seed.models import PropertyState
PropertyState.objects.get(pm_property_id=<PM Property ID of your choice>).propertyview_set.first().property.updated 
>>> old time 
```
inventory > select property with same PM Property ID > actions > refresh metadata ...
... back to shell 
```
PropertyState.objects.get(pm_property_id=<Same PM Property ID>).propertyview_set.first().property.updated
>>> current time
```

#### What are the relevant tickets?
#3218 

#### Screenshots (if appropriate)

![Screen Shot 2022-04-28 at 1 47 39 PM](https://user-images.githubusercontent.com/58446472/165833833-c2766c63-4fa8-45fd-b870-ef0d7aacda2b.png)
![Screen Shot 2022-04-28 at 1 47 31 PM](https://user-images.githubusercontent.com/58446472/165833835-dd88c4b3-0280-4f08-adb6-245c0ce682da.png)
![Screen Shot 2022-04-28 at 1 47 21 PM](https://user-images.githubusercontent.com/58446472/165833830-b40b9ff8-fee9-4f04-9209-ae13266f4420.png)
